src/Module/Conversation/Community.php

```diff
@@ -0,0 +1,334 @@
+
+			if (!$available) {
+				$available = (self::$page_style == CP_USERS_ON_SERVER) && (self::$content == 'local');
+			}
+
+			if (!$available) {
+				$available = (self::$page_style == CP_GLOBAL_COMMUNITY) && (self::$content == 'global');
+			}
+
+			if (!$available) {
+				throw new HTTPException\ForbiddenException(DI::l10n()->t('Not available.'));
+			}
+		}
+
+		// check if we serve a mobile device and get the user settings accordingly
+		if (DI::mode()->isMobile()) {
+			self::$itemsPerPage = DI::pConfig()->get(local_user(), 'system', 'itemspage_mobile_network', 20);
+		} else {
+			self::$itemsPerPage = DI::pConfig()->get(local_user(), 'system', 'itemspage_network', 40);
+		}
+
+		// now that we have the user settings, see if the theme forces
+		// a maximum item number which is lower then the user choice
+		if ((DI::app()->force_max_items > 0) && (DI::app()->force_max_items < self::$itemsPerPage)) {
+			self::$itemsPerPage = DI::app()->force_max_items;
+		}
+
+		self::$since_id = $_GET['since_id'] ?? null;
+		self::$max_id   = $_GET['max_id']   ?? null;
+	}
+
+	/**
+	 * Computes the displayed items.
+	 *
+	 * Community pages have a restriction on how many successive posts by the same author can show on any given page,
+	 * so we may have to retrieve more content beyond the first query
+	 *
+	 * @return array
+	 * @throws \Exception
+	 */
+	protected static function getItems()
+	{
+		$items = self::selectItems(self::$since_id, self::$max_id, self::$itemsPerPage);
+
+		$maxpostperauthor = (int) DI::config()->get('system', 'max_author_posts_community_page');
+		if ($maxpostperauthor != 0 && self::$content == 'local') {
+			$count = 1;
+			$previousauthor = '';
+			$numposts = 0;
+			$selected_items = [];
+
+			while (count($selected_items) < self::$itemsPerPage && ++$count < 50 && count($items) > 0) {
+				foreach ($items as $item) {
+					if ($previousauthor == $item["author-link"]) {
+						++$numposts;
+					} else {
+						$numposts = 0;
+					}
+					$previousauthor = $item["author-link"];
+
+					if (($numposts < $maxpostperauthor) && (count($selected_items) < self::$itemsPerPage)) {
+						$selected_items[] = $item;
+					}
+				}
+
+				// If we're looking at a "previous page", the lookup continues forward in time because the list is
+				// sorted in chronologically decreasing order
+				if (isset(self::$since_id)) {
+					self::$since_id = $items[0]['commented'];
+				} else {
+					// In any other case, the lookup continues backwards in time
+					self::$max_id = $items[count($items) - 1]['commented'];
+				}
+
+				$items = self::selectItems(self::$since_id, self::$max_id, self::$itemsPerPage);
+			}
+		} else {
+			$selected_items = $items;
+		}
+
+		return $selected_items;
+	}
+
+	/**
+	 * Database query for the comunity page
+	 *
+	 * @param $since_id
+	 * @param $max_id
+	 * @param $itemspage
+	 * @return array
+	 * @throws \Exception
+	 */
+	private static function selectItems($since_id, $max_id, $itemspage)
```

Comment:
New Model at `src\Model\Item\Community` with having a method `select()` ?
=> At a later stage, it's more a repository call for returning `item`s, but not yet :)

Reply:
I'd rather directly jump to a repository than have the intermediate static model class that will be replaced as well eventually. But this really isn't the point of this PR.

Reply:
Hm.. Ok at least mark it for later :)